repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -231,6 +288,23 @@ def _prep_repository_access(context, target_userspace):
         run(['rm', '-rf', os.path.join(target_etc, 'rhsm')])
         context.copytree_from('/etc/pki', os.path.join(target_etc, 'pki'))
         context.copytree_from('/etc/rhsm', os.path.join(target_etc, 'rhsm'))
+
+    # Copy RHN data independent from RHSM config
+    if os.path.isdir('/etc/sysconfig/rhn'):
+        context.call(['/usr/sbin/rhn_check'], callback_raw=utils.logging_handler)
+        run(['rm', '-rf', os.path.join(target_etc, 'sysconfig/rhn')])
```

Comment:
`shutil.rmtree(os.path.join(target_etc, 'sysconfig/rhn'))` could be slightly faster